code/modules/hydroponics/grown.dm

```diff
@@ -897,7 +897,7 @@ obj/item/weapon/reagent_containers/food/snacks/grown/shell/eggy/add_juice()
 
 /obj/item/weapon/reagent_containers/food/snacks/grown/tomato/blue/bluespace/attack_self(var/mob/user)
 	squish(user)
-	user.drop_item()
+	user.unEquip(src)

```

Comment:
You don't need to drop a tomato to squish it, that's why I didn't bother with an if.